websockets/client.py

```diff
@@ -35,9 +35,8 @@ def write_http_request(self, path, headers):
 
         """
         self.path = path
-        self.request_headers = email.message.Message()
-        for name, value in headers:
-            self.request_headers[name] = value
+        self.request_headers = http.client.HTTPMessage()
+        self.request_headers._headers = headers     # HACK
```

Comment:
To start with, I think it could be as simple as a class with two attributes for "raw" and "not raw" (matching the two attributes currently stored on the Protocol classes), and you wouldn't lose anything. And then you could add whatever helper methods as needed, like the get / set header functions.

Later on (and only if you wanted), you could get more sophisticated with how you store and mutate the raw header info internally to the class.